### PR TITLE
cleanup levenshtein and fix cpplint warnings

### LINKED ISSUE #44: Cleanup Levenshtein class code

### PR DESCRIPTION
**Pull request by AndreasDahl**
fixes #44 

Please let me know if there is anything more to be done.

Note that i deliberately did not split the function into sub-functions. I strongly agree with what @Chrikomu said at our meeting about not leaving the scope of a [`new`](https://github.com/BIO-DIKU/klust/compare/cleanup-levenshtein?expand=1#diff-f831fa14dc7a69883d32b1d8942e5298R36) until it has been [`delete`d](https://github.com/BIO-DIKU/klust/compare/cleanup-levenshtein?expand=1#diff-f831fa14dc7a69883d32b1d8942e5298R87) if possible. If you disagree i will be happy to pick up the discussion.

### CODE REVIEW

**Comment by Chrikomu** [comment]
Please use `#include <cctype>` in levenshtein.cc, which is where `tolower(int)` is defined.

File: src/modules/levenshtein.cc [hunk 1]
```diff
@@ -20,52 +20,55 @@
...
+        ? str2.length() - str1.length()
+        : str1.length() - str2.length());
+  size_t smallestLength = (str1.length() < str2.length()
+        ? str1.length()
+        : str2.length());
   // Maximal errors before fail-fast kicks in
   float maxErrors = getIdentity() * smallestLength + lengthDiff;
 
+  // degenerate cases
+  if (str1.length() == 0 && maxErrors < str2.length()) return false;
```
Comment (Chrikomu) [inline src/modules/levenshtein.cc]: Delete v0 and v1!

File: src/modules/levenshtein.cc [hunk 2]
```diff
@@ -20,52 +20,55 @@
...
+  if (str2.length() == 0 && maxErrors < str1.length()) return false;
```
Comment (Chrikomu) [inline src/modules/levenshtein.cc]: And same here.

**Comment by AndreasDahl** [comment]
I can't stop commiting! :grimacing:

**Pull request merged**